flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/result/ResultFetcher.java

```diff
@@ -134,9 +138,22 @@ private ResultFetcher(
     }
 
     public static ResultFetcher fromTableResult(
-            OperationHandle operationHandle,
-            TableResultInternal tableResult,
-            boolean isQueryResult) {
+            OperationHandle operationHandle, TableResult tableResult, boolean isQueryResult) {
+        CloseableIterator<RowData> resultRows;
+        RowDataToStringConverter rowDataToStringConverter;
+        if (tableResult instanceof TableResultInternal) {
+            TableResultInternal tableResultInternal = (TableResultInternal) tableResult;
+            resultRows = tableResultInternal.collectInternal();
+            rowDataToStringConverter = tableResultInternal.getRowDataToStringConverter();
+        } else {
+            // sometimes, the tableResult maybe not an instance of TableResultInternal
+            // in the case of pluggable dialect implements method
+            // ExtendedOperationExecutor#executeOperation, and it doesn't return TableResultInternal
```

Comment:
Which case doesn't return `TableResultInternal`?

Reply:
Previous, we have `ExtendedOperationExecutor#executeOperation` marked as `PublicEvolving`, so the result of this method should be `TableResult` which is also marked as `PublicEvolving`.  Since this interface don't require implementer to return `TableResultInternal`, the implementer can return anything that implements `TableResult`.

But now, I have removed `ExtendedOperationExecutor` and reverted this change.